tfx/components/trainer/executor.py

```diff
@@ -206,13 +214,15 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
     # Train the model
     absl.logging.info('Training model.')
     run_fn(fn_args)
-
+    
     # Note: If trained with multi-node distribution workers, it is the user
     # module's responsibility to export the model only once.
     if not tf.io.gfile.exists(fn_args.serving_model_dir):
       raise RuntimeError('run_fn failed to generate model.')
-    absl.logging.info('Training complete. Model written to %s',
-                      fn_args.serving_model_dir)
+    if not tf.io.gfile.exists(fn_args.model_run_dir):
+      raise RuntimeError('run_fn failed to write logs to correct location.')
```

Comment:
let's allow empty for now, users may not need tensorboard

in the future we can consider make this an optional output artifact